### PR TITLE
Adding geneve support in p4_ovs

### PR DESCRIPTION
**Pull request by satish153**
Added tunnel_type field to identify tunnel type.

### CODE REVIEW

**Review by ffoulkes — CHANGES_REQUESTED** [review]
You need to sign your commits by specifying the `-s` parameter when you run `git commit`. Failure to do so results in a DCO error.

This needs to be done for every commit in the PR.

I believe you can fix this after the fact by doing the following:
```bash
git commit -s --amend
(:wq to save updated commit comment)
git push -f
```

**Review by ffoulkes — APPROVED** [review]
LGTM

File: include/openvswitch/ovs-p4rt.h [hunk 1]
```diff
@@ -35,6 +35,31 @@ enum p4_vlan_mode {
   P4_PORT_VLAN_UNSUPPORTED
 };
 
+enum p4_tunnel_type {
+  NO_MODIFY,
+  VXLAN_ENCAP,
+  VXLAN_DECAP_OUTER_HDR,
+  NEIGHBOR,
+  VLAN_PUSH,
+  VLAN_POP,
```
Comment (nupuruttarwar) [inline include/openvswitch/ovs-p4rt.h]: why are we considering VLAN_PUSH, VLAN_POP, NEIGHBOR, etc as p4_tunnel_type? Should this be only vxlan and geneve and similar to ovs_tunnel_type declared below? In OVS, I see following tunnel configs in one of the functions

    return (class->get_config == get_tunnel_config &&
            (!strcmp("geneve", type) || !strcmp("vxlan", type) ||
             !strcmp("lisp", type) || !strcmp("stt", type) ||
             !strcmp("gtpu", type) || !strcmp("bareudp",type)));
}

I understand we are supporting vxlan and geneve out of the above

Comment (satish153) [reply include/openvswitch/ovs-p4rt.h]: Thanks for noticing this.
These are actually mod profiles.
My initial logic was to have common action and different action params for each of mod profiles. So, I created this enum to map to mod profiles in the lnw.p4 program. Later, I had to change that logic, as mod action cannot be set runtime.
This enum is not needed anymore.

**Review by 5abeel — APPROVED** [review]
LGTM

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Comment by nupuruttarwar** [comment]
@satish153 please sign off your commit for DCO check to pass

**Pull request merged**